### PR TITLE
adds -output-all flag

### PR DESCRIPTION
**Pull request by wassup05**
First of all, thanks for this amazing tool.

The PR Addresses #111 

implemented printing all the textual contetnts of clipboard in two ways `raw` (using the %q specifier) and `unescaped` (printing contents as it is)
each entry printed is separated by a newline

Also adds a basic `.gitignore`

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -184,3 +188,19 @@ func handleForceClose() {
...
+		for _, v := range items {
+			fmt.Printf("%q\n", v.Value)
+		}
+	} else if format == "unescaped" {
+		for _, v := range items {
+			fmt.Println(v.Value)
+		}
+	} else {
+		fmt.Printf("Invalid argument to -output-all\nSee %s --help for usage", os.Args[0])
+	}
```
Comment (ccoVeille) [inline main.go:205]: This could be a `switch format `

File: main.go [hunk 2]
```diff
@@ -184,3 +188,19 @@ func handleForceClose() {
...
+	if format == "raw" {
+		for _, v := range items {
+			fmt.Printf("%q\n", v.Value)
+		}
+	} else if format == "unescaped" {
+		for _, v := range items {
+			fmt.Println(v.Value)
+		}
+	} else {
+		fmt.Printf("Invalid argument to -output-all\nSee %s --help for usage", os.Args[0])
```
Comment (ccoVeille) [inline main.go:204]: utils.LogError is also used for other error reporting

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+clipse
```
Comment (ccoVeille) [inline .gitignore:1]: It's strange to see this committed, ay least with changes that are unrelated

File: main.go (hunk 2 above)
Comment (wassup05) [reply main.go:204]: utils.LogError logs to a file, I don't think that's the thing you would want to do if it's an error concerning command line arguments

File: .gitignore (hunk 3 above)
Comment (wassup05) [reply .gitignore:1]: Oh yes, sorry I should have mentioned that... I thought it would be useful as anyone contributing would be building clipse regularly and it would be a good convenience to have

**Review by ccoVeille — APPROVED** [review]

**Comment by savedra1** [comment]
Hi @wassup05, 

Thank you so much for this contribution :blue_heart: I have tested this and it works really well. I have one thing to get your thoughts on, I'll add my comment to the open review ( thanks @ccoVeille ).

File: main.go (hunk 1 above)
Comment (savedra1) [reply main.go:205]: @wassup05 As we wouldn't expect further options here, I'm wondering if there's an elegant way to drop the need for specifying the "unescaped" value and use that as the default if the `-output-all` flag is supplied, with the option to add "raw" if needed. Just thinking that could be a simpler implementation for both the UX and the code. 

I realise this isn't possible with a string flag in Go, as a value would always be needed there, but maybe setting this to a bool and checking for the "raw" arg explicitly could be a simple way to achieve this. What do you think?

Comment (savedra1) [reply main.go:205]: E.g. 

```go

outputAll   = flag.Bool("output-all", false, "Print clipboard text content to stdout, each entry separated by a newline. Add the value 'raw' for escaped output.")
extraArgVal = flag.Arg(2)

...

case *outputAll:
	handleOutputAll()

...

func handleOutputAll() {
	items := config.TextItems()

	if extraArgVal == "raw" {
		for _, v := range items {
			fmt.Printf("%q\n", v.Value)
		}
		return
	}

	for _, v := range items {
		fmt.Println(v.Value)
	}
}
```

Comment (wassup05) [reply main.go:205]: Hi @savedra1, Thanks for reviewing
Although I see how that would be useful, It could bring some inconsistencies into pitcure like `flag` supports 4 formats iirc,

`-flag a`
`-flag=a`
`--flag a`
`--flag=a`

But we wouldn't be able to do that if we make it a `flag.Bool`

Another alternative would be to set the `flag.String` default value as something different from "raw" or "unescaped" like "notset" and then we see if this has been changed by the user... But I see a drawback with this too which being that the user may be misguided by the default value printed by `flag.PrintDefaults`

What do you think?

Comment (savedra1) [reply main.go:205]: Hi @wassup05, apologies for such a delayed response here. 

I take your point, I think your approach works well as it is, so there's no need to overthink. 

I will be merging this now 👍 

Thank you again for your work here.
